### ISSUE TITLE
Script coverage stops at << in bash script

### ISSUE THREAD

**Issue by philwhineray** [SimonKagstrom/kcov#116]
Below are a couple of examples, in both cases the coverage stops at the line `<<` for me with version 29, also with git master (do you have a preference for bug reports?).

~~~~
$ bash --version
GNU bash, version 4.3.30(1)-release (x86_64-pc-linux-gnu)
~~~~

Arithmetic:

~~~~
#!/bin/bash

echo "Start"

SUB=30
max=$[ 1 << (32 - SUB) ]

echo "$max"

echo "Not covered"
echo "Not covered"
echo "Not covered"
~~~~

Heredoc:

~~~~
#!/bin/bash

echo "Start"

cat <<-!
	Middle
!

echo "Not covered"
echo "Not covered"
echo "Not covered"
~~~~

I have seen other heredoc examples that work, so I guess it may be the use of '!' as the EOF, or the '-' that are causing that problem in that script

**Comment by SimonKagstrom** [comment]
Issues are perfect for bug reports, so that's fine!

The heredoc problem should be found somewhere around here:

  https://github.com/SimonKagstrom/kcov/blob/master/src/engines/bash-engine.cc#L571

I'm not sure what causes it though. As for the arithmetic problem, I think it's because I don't check for $[, and that you have a shift-left operator in your example. That should be around here:

  https://github.com/SimonKagstrom/kcov/blob/master/src/engines/bash-engine.cc#L588


Anyway, I'm not proud of the bash scanning code. It should really be replaced by a proper parser, but I don't know bash well enough to write one myself, and haven't found a good and simple example to copy from.

**Comment by SimonKagstrom** [comment]
Fixed with fff5ad9 and merge #117 .
